### PR TITLE
Carabiner

### PR DESCRIPTION
**Pull request by Kurfursten**
The only thing spookier than me reviving this months-old project is that I properly squished the billion commits it has had over time.
- Carabiner, available from autolathe
- Hooks onto specific objects (Allowed: mopbucket, roller bed, crate, air pump, scrubber, space heater, air conditioner, canister)
- These are hooked to you in a more robust way that prevents being pushed apart in hallways
- You can still disarm to force a release of the pull
- If you can't pull it (it's anchored, there's something in the way, etc.) it pops off
- It pops off if you drop or put away the carabiner, but not if you put it in your pocket
- Fits on a toolbelt

Fully tested

untitled

### CODE REVIEW

**Comment by Shadowmech88** [comment]
Janitors might like it if they could clip their mop bucket with a carabiner.

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 1]
```diff
@@ -39,6 +39,14 @@
 		r_store,
 		w_uniform)
 
+/mob/living/carbon/human/get_nonclothing_items() //pockets, hands, belt
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/human/inventory.dm]: What about backpack?

Comment (ComicIronic) [inline code/modules/mob/living/carbon/human/inventory.dm]: ALSO WHY

Comment (Kurfursten) [inline code/modules/mob/living/carbon/human/inventory.dm]: This is the inverse of get_clothing_items() which gets everything except those three types

Comment (ComicIronic) [inline code/modules/mob/living/carbon/human/inventory.dm]: So then this should just be get_all_slots() - get_clothing_items().
It's also not used anywhere.

Comment (Kurfursten) [inline code/modules/mob/living/carbon/human/inventory.dm]: Isn't it more efficient to just refer to the slots than to make two calls?

Comment (ComicIronic) [inline code/modules/mob/living/carbon/human/inventory.dm]: It's more efficient but more redundant, and therefore harder to maintain.
The cost isn't that much higher, and it's not like inventory code is called every tick.

Comment (Kurfursten) [inline code/modules/mob/living/carbon/human/inventory.dm]: It won't matter, I'm taking it out in the next revision

There was a more efficient method to get what I wanted.

**Comment by Kurfursten** [comment]
Got a new sprite, now can handle mop buckets, and I got rid of that unnecessary nonclothing proc.

Ready for NFD.

**Comment by Shadowmech88** [comment]
For what it's worth, it irks me that the carabiner's hinge is on the larger end.

**Comment by ComicIronic** [comment]
There's too much snowflake in this.
You should be using some sort of pull handler for how strong a pull is, instead of a specific typed var that's harder to maintain and keep sane.

**Comment by Kurfursten** [comment]
The specific typed var is necessary because you need to be able to handle references back to the carabiner if the pull is broken. This is faster than looping through the slots to find each carabiner and making sure it's open.

**Comment by ComicIronic** [comment]
Faster != better.

**Comment by Kurfursten** [comment]
Are you sure that's what you want? This is definitely the fastest way for performance, and faster is often better.

**Comment by ComicIronic** [comment]
Faster is better within a good system.

**Comment by Kurfursten** [comment]
Goodness is arbitrary if not based on a measurable quantity

**Pull request closed without merging**